examples/run.rs

```diff
@@ -0,0 +1,87 @@
+use async_trait::async_trait;
+use faktory::{Client, Job, JobRunner, Worker};
+use std::io::{Error as IOError, Result as IOResult};
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use tokio::time;
+
+pub struct JobHandler<T> {
+    chan: Arc<mpsc::Sender<T>>,
+}
+
+impl<T> JobHandler<T> {
+    pub fn new(chan: &Arc<mpsc::Sender<T>>) -> Self {
+        Self {
+            chan: Arc::clone(chan),
+        }
+    }
+}
+
+impl JobHandler<u64> {
+    async fn process_one(&self, job: Job) -> IOResult<()> {
+        time::sleep(time::Duration::from_millis(100)).await;
+        let mut args = job.args().into_iter();
```

Comment:
Why go via an iterator instead of just directly indexing into `job.args()`?

Reply:
I guess I just applied the same pattern as with CLI args, they look identical. This encourages us to handle the `None` case and helps avoid out of range indexing. But for this example's sake, we can simplify things of course and index directly into the `job.args()`.

Reply:
Updated